tests/functests/instancetype_test.go

```diff
@@ -98,15 +98,25 @@ var _ = Describe("Common instance types func tests", func() {
 			Entry("[test_id:10743] Ubuntu 22.04", ubuntu2204ContainerDisk, "ubuntu", "ubuntu", false),
 		)
 
-		DescribeTable("a Windows guest with", func(containerDisk, preference, username, password string) {
+		DescribeTable("a Windows guest with", func(containerDisk, preference, username, password string, guestAgent, ssh bool) {
 			vm = randomVM(&v1.InstancetypeMatcher{Name: "u1.large"}, &v1.PreferenceMatcher{Name: preference}, true)
 			addContainerDisk(vm, containerDisk)
 			vm, err = virtClient.VirtualMachine(testNamespace).Create(context.Background(), vm)
 			Expect(err).ToNot(HaveOccurred())
 			expectVMToBeReady(virtClient, vm.Name)
-			expectSSHToRunCommandWithPassword(virtClient, vm.Name, username, password)
+			if guestAgent {
+				expectGuestAgentToBeConnected(virtClient, vm.Name)
+			}
+			if ssh {
+				expectSSHToRunCommandWithPassword(virtClient, vm.Name, username, password)
+			}
 		},
-			Entry("[test_id:10739] Validation OS", validationOsContainerDisk, "windows.11", "Administrator", "Administrator"),
+			Entry("[test_id:10739] Validation OS", validationOsContainerDisk, "windows.11", "Administrator", "Administrator", false, true),
+			Entry("[test_id:????] Windows 10", windows10ContainerDisk, "windows.10", "Administrator", "Administrator", true, true),
+			Entry("[test_id:????] Windows 11", windows11ContainerDisk, "windows.11", "Administrator", "Administrator", true, true),
+			Entry("[test_id:????] Windows Server 2016", windows2k16ContainerDisk, "windows.2k16", "Administrator", "Administrator", true, false),
+			Entry("[test_id:????] Windows Server 2019", windows2k19ContainerDisk, "windows.2k19", "Administrator", "Administrator", true, true),
+			Entry("[test_id:????] Windows Server 2022", windows2k22ContainerDisk, "windows.2k22", "Administrator", "Administrator", true, true),
```

Comment:
Does it make sense to cover all `guestAgent` and `ssh` combinations?

Reply:
No, these are specific to the images.

Reply:
Got it!